docs/content/latest/explore/ysql-language-features/partitions.md

```diff
@@ -38,23 +38,23 @@ A regular table cannot become a partitioned table, just as a partitioned table c
 Suppose you work with a database that includes the following table:
 
 ```sql
-CREATE TABLE employees (
-  employee_no integer PRIMARY KEY,
-  name text,
-  department text,
-  change_date date
+CREATE TABLE product_changes (
+  product_name text,
+  price int,
+  change_date date PRIMARY KEY
 );
 ```
 
-*change_date* represents the date when any type of change occurred on the employee record. This date is typically required when generating monthly reports. Assuming that typically only the last month's data is queried often, then the data that is older than one year is removed from the table every month. To simplify this process, you can partition the `employees` table. You start by specifying bounds corresponding to the partitioning method and partition key of the `employees` table. This means you create partitions as regular tables and YSQL generates partition constraints automatically based on the partition bound specification every time they have to be referenced.
+*change_date* represents the date when any type of change occurred on the product record. This date might be required when generating monthly reports. Assuming that typically only the last month's data is queried often, then the data older than one year is removed from the table every month. To simplify this process, you can partition the `product_changes` table. You start by specifying bounds corresponding to the partitioning method and partition key of the `product_changes` table. This means you create partitions as regular tables and YSQL generates partition constraints automatically based on the partition bound specification every time they have to be referenced.
+
+Note that the primary key for a table should always contain partition key. In the current release of YugabyteDB, it is recommended to set the primary key directly on partitions.
 
-You can declare partitioning on the `employees` table by first creating it as a partitioned table. You specify the `PARTITION BY` clause and supply it with the partitioning method such as `RANGE`, as well as a list of columns as a partition key, as shown in the following example:
+You can declare partitioning on a table by creating it as a partitioned table: you specify the `PARTITION BY` clause which you supply with the partitioning method, such as `RANGE`, and a list of columns as a partition key, as shown in the following example:
 
 ```sql
-CREATE TABLE employees (
-  employee_no integer,
-  name text,
-  department text,
+CREATE TABLE product_changes (
+  product_name text,
+  price int,
   change_date date PRIMARY KEY
```

Comment:
Remove this here and above. Its super weird to have change_date as PRIMARY_KEY

Reply:
Please can you come up with a different application or use the one I pointed above that @rkarthik007 used? There the primary key was a function of the partition key in an application that actually requires it. Here it looks a bit weird actually to think that a product will change only once a day...

Reply:
Also I thought you were going to add the link to row level geo partitioning - I could not find it..

Reply:
Added the link